File: Storage/Code/231.2-的幂.py
#
# @lc app=leetcode.cn id=231 lang=python3
#
# [231] 2 的幂
#

# @lc code=start
class Solution:
    def isPowerOfTwo(self, n: int) -> bool:
        if n<=0:
            return False
        if n==1:
            return True
        while n>1:
            if n%2!=0:
                return False
            n=n/2
        return True

# @lc code=end

File: Storage/Code/231.2-的幂.cpp
/*
 * @lc app=leetcode.cn id=231 lang=cpp
 *
 * [231] 2 的幂
 */

// @lc code=start
class Solution {
public:
    bool isPowerOfTwo(int n) {
        if(n<=0)    return false;
        if(n==1)    return true;
        while(n!=1){
            if(n%2!=0)  return false;
            n=(n>>1);
        }
        return true;
    }
};
// @lc code=end

